lblod/ember-rdfa-editor
# GN-5116: `block_rdfa` redesign

### Overview
This PR reworks the UX/UI of the `block_rdfa` editor node.
Specifically, it includes the following adjustments:
- Adjustments to the margins and borders of `block_rdfa` nodes
- Introduction of the `BlockRDFaView` nodeview
- Addition of a descriptive `label` component to the `block_rdfa` view.
- Simplification and margin adjustments of the `gap-cursor` styling
- Addition of `namespace` utils

`block_rdfa` nodes now contain a label which is shown in editor. This label is determined using the following priority:
1. Usage of the `label` attribute
2. Usage of the `skos:prefLabel` attribute (if the node is `rdfa-aware` and a resource-node)
3. Last part/fragment of type/predicate URI (if the node is `rdfa-aware`)
4. A fallback string

**Before**
![image](https://github.com/user-attachments/assets/155038be-1aed-4b64-a264-23e8045da0c4)

**After**
![image](https://github.com/user-attachments/assets/148d347d-c89c-4aed-a575-4299a895fe05)


##### connected issues and PRs:
[GN-5116](https://binnenland.atlassian.net/browse/GN-5116?atlOrigin=eyJpIjoiOWE3NzM0MTNmOGE1NDgxYjkwNTg1ZDZhNGQ2MWVkYzYiLCJwIjoiaiJ9)

### How to test/reproduce
- Start the dummy app
- Open the `editable nodes (POC)` page
- Ensure the different blocks now show a label
- Add a `skos:prefLabel` property to a node
- Ensure that the label shown changes
- Edit the `label` attribute using the 'Node attributes' pane, the label shown should change again

### Challenges/uncertainties
- There is no support for internationalization (yet)
- I have not yet added a seperate component to edit the `label` attribute, not sure if desired
- Not completely sure about the margins, did them best-effort
- Is the way I styled the label of the blocks the best way (using `position: absolute`)?

### Checks PR readiness
